src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    public static final String DELIMITER = ",";
+    public static final int OPERATION_INDEX = 0;
+    public static final int AMOUNT_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyTotalAmount = 0;
+        int buyTotalAmount = 0;
+        for (String line : readFromFile(fromFileName).split(System.lineSeparator())) {
+            String[] row = line.split(DELIMITER);
+            if (row[OPERATION_INDEX].equals("supply")) {
+                supplyTotalAmount += Integer.valueOf(row[AMOUNT_INDEX]);
+            } else {
+                buyTotalAmount += Integer.valueOf(row[AMOUNT_INDEX]);
+            }
+        }
+        StringBuilder result = new StringBuilder();
+        result.append("supply,").append(supplyTotalAmount).append(System.lineSeparator())
+                .append("buy,").append(buyTotalAmount).append(System.lineSeparator())
+                .append("result,").append((supplyTotalAmount - buyTotalAmount));
+        printToFile(toFileName, result.toString());
```

Comment:
```suggestion
        String dataFromFile = readFromFile(fromFileName);
        String report = createReport(dataFromFile);
        printToFile(toFileName, report);
```
the whole logic is ok, just move it to a separate method